artifact-writer/app/main.py

```diff
@@ -0,0 +1,95 @@
+import os
+import logging
+from typing import Optional, List
+from azure.storage.blob import BlobServiceClient, StorageErrorCode
+
+
+logging.basicConfig(level=logging.INFO, format='%(name)s | %(levelname)s | %(message)s')
+logging.getLogger('azure.core.pipeline.policies.http_logging_policy').setLevel(logging.WARNING)
+logger = logging.getLogger(__name__)
+
+
+try:
+    from dotenv import load_dotenv
+    load_dotenv()
+except ModuleNotFoundError:
+    logger.info("Failed to load dotenv file. Assuming production")
+
+
+def find_artifacts(location: str) -> List[str]:
+    artifacts = []
+    for root, dirs, files in os.walk(location):
+        for file_ in files:
+            artifacts.append(os.path.join(root, file_))
+    logger.info(f'Found artifacts: {artifacts}')
+    return artifacts
+
+
+def log_env(artifact_location: str, target_bucket: str, target_folder: str):
+    logger.info(f'Artifact location:    {artifact_location}')
+    logger.info(f'Target bucket:        {target_bucket}')
+    logger.info(f'Target folder:        {target_folder} (if None, becomes root of bucket)')
+
+
+def upload_file(file_: str, target_blob_name: str, target_bucket: str, blob_servie_client: BlobServiceClient):
+    logger.info(f'Uploading {file_} ({os.path.getsize(file_)} B) to {target_bucket} at {target_blob_name}')
+    blob_client = blob_servie_client.get_blob_client(container=target_bucket, blob=target_blob_name)
```

Comment:
```suggestion
def upload_file(file_: str, target_blob_name: str, target_bucket: str, blob_service_client: BlobServiceClient):
    logger.info(f'Uploading {file_} ({os.path.getsize(file_)} B) to {target_bucket} at {target_blob_name}')
    blob_client = blob_service_client.get_blob_client(container=target_bucket, blob=target_blob_name)
```

`blob_servie_client` -> `blob_service_client`